suite/tests/linux/clone.c

```diff
@@ -120,23 +121,23 @@ test_thread(bool share_sighand, bool clone_vm, bool use_clone3)
 int
 main()
 {
-    /* Try using clone3 when it is possibly not defined. */
+    /* Try using clone3 when it is possibly not defined. This is done for two
+     * reasons: test whether the kernel supports it, and our handling of clone3
+     * when it doesn't.
+     */
     int ret_failure_clone3 = make_clone3_syscall(NULL, 0, NULL);
     assert(ret_failure_clone3 == -1);
 
-#ifndef ANDROID /* clone3 not supported on Android */
-    /* i#6596 In some scenarios SYS_clone3 is defined but clone3 returns ENOSYS
-     * e.g. when running in a container under Ubuntu 22.04
+    /* In some environments, we see that the kernel supports clone3 even though
+     * SYS_clone3 is not defined by glibc. So we don't predicate our efforts on
+     * whether SYS_clone3 is defined.
+    /* Plus in some scenarios SYS_clone3 is defined but clone3 returns ENOSYS.
```

Comment:
Since this "Plus..." is a continuation of the above statement, better to not keep it in a separate comment, otherwise it loses context.

Reply:
Done